package-build.el

```diff
@@ -1080,9 +1094,10 @@ and a cl struct in Emacs HEAD.  This wrapper normalises the results."
 
     (package-build--message "\n;;; %s\n" file-name)
     (let* ((version (package-version-join
-                     (version-to-list
+                     (package-build--version-to-list
```

Comment:
I don't think we need `package-build--version-to-list` here either. Because the *checkout* should return us a valid version and we should error if it doesn't. Because that means the regexp is broken.